pydoctor/test/test_commandline.py

```diff
@@ -59,3 +63,45 @@ def test_cache_disabled_by_default() -> None:
     parser = driver.getparser()
     (options, _) = parser.parse_args([])
     assert not options.enable_intersphinx_cache
+
+
+def test_cli_warnings_on_error() -> None:
+    """
+    The -W option is disabled by default.
+    """
+    options, args = driver.parse_args([])
+    assert options.warnings_as_errors == False
+
+    options, args = driver.parse_args(['-W'])
+    assert options.warnings_as_errors == True
+
+
+def test_main_return_zero_on_warnings() -> None:
+    """
+    By default it will return 0 as exit code even when there are warnings.
+    """
+    output = StringIO()
+    with redirect_stdout(output):
+        exit_code = driver.main(args=[
+            '-v', '--testing',
+            'pydoctor/test/testpackages/basic/'
+            ])
+
+    assert exit_code == 0
+    assert '<None> guessing basic for project name' in output.getvalue()
```

Comment:
The `<None>` doesn't look great; the user might think they hit a bug.

In general, I also prefer quotes around names (`guessing "basic" for`), so it's easier for the user to see which parts of the message are static and which are dynamic. I know this message isn't new, but if we're going to test the exact output it produces, now would be a good time to update it.

I notice that in the pydoctor code, sometimes inserted strings are not quoted, sometimes they are quoted in single quotes and sometimes in double quotes. It would be good to pick a single style and stick to it. I have a strong preference for quoted, but no real preference for single vs double quotes; what do you think?

Reply:
+1 for the quotes.

Any suggestion for a replacement for "<None>" ?